src/hotspot/share/opto/graphKit.cpp

```diff
@@ -1187,7 +1187,9 @@ Node* GraphKit::load_array_length(Node* array) {
     alen = alloc->Ideal_length();
     Node* ccast = alloc->make_ideal_length(_gvn.type(array)->is_oopptr(), &_gvn);
     if (ccast != alen) {
-      alen = _gvn.transform(ccast);
+      _gvn.set_type_bottom(ccast);
+      record_for_igvn(ccast);
+      alen = ccast;
```

Comment:
@vnkozlov  Thanks for your review!

GraphKit::load_array_length code is different with new_array() and arraycopy_move_allocation_here(), it doesn't replace_in_map. Currently GraphKit::cast_replace_array_length_post_allocation warpper is used in new_array() and arraycopy_move_allocation_here().

Would you prefer a wrapper for all these three places? It would be like following code.
```
Node* GraphKit::cast_array_length_post_allocation(AllocateArrayNode* alloc, const TypeOopPtr* oop_type, bool replace_map /false in load_array_length/) {
  Node* length = alloc->in(AllocateNode::ALength);
  if (replace_map == false || map()->find_edge(length) >= 0) {
    Node* ccast = alloc->make_ideal_length(oop_type, &_gvn);
    if (ccast != length) {
      _gvn.set_type_bottom(ccast);
      record_for_igvn(ccast);
      if (replace_map) {
        replace_in_map(length, ccast);
      }
      return cast;
    }
  }
  return length;
}
```

Reply:
All places, please. I prefer different names for this method and its parameters `GraphKit::array_ideal_length(alloc , ary_type, replace_length_in_map)`. No need for parameter's comment in declaration - it is clear from name.

Reply:
done， new wrapper is added, all 3 places: 1) new_array 2) arraycopy_move_allocation_here 3)  load_array_length  use GraphKit::array_ideal_length(alloc , ary_type, replace_length_in_map)